src/tryceratops/analyzers/conditional.py

```diff
@@ -0,0 +1,62 @@
+import ast
+from typing import Union
+
+from tryceratops.violations import codes
+
+from .base import BaseAnalyzer
+
+STANDARD_NON_TYPE_ERROR_IDS = (
+    "ArithmeticError",
+    "AssertionError",
+    "AttributeError",
+    "BufferError",
+    "EOFError",
+    "Exception",
+    "ImportError",
+    "LookupError",
+    "MemoryError",
+    "NameError",
+    "ReferenceError",
+    "RuntimeError",
+    "SyntaxError",
+    "SystemError",
+    "ValueError",
+)
+
+
+class PreferTypeErrorAnalyzer(BaseAnalyzer):
+    violation_code = codes.PREFER_TYPE_ERROR
+
+    def _is_checking_type(self, node: Union[ast.stmt, ast.expr]) -> bool:
+        if isinstance(node, ast.If):
+            return self._is_checking_type(node.test)
+        if isinstance(node, ast.UnaryOp):
+            return self._is_checking_type(node.operand)
+        if isinstance(node, ast.BoolOp):
+            for value in node.values:
+                if self._is_checking_type(value):
+                    return True
+        if isinstance(node, ast.Call):
+            if isinstance(node.func, ast.Name):
+                if node.func.id in ("isinstance", "issubclass", "callable"):
+                    return True
+        return False
+
+    def _check_is_raise_other_than_typeerror(self, node: ast.Raise) -> None:
+        if isinstance(node.exc, ast.Call):
+            if isinstance(node.exc.func, ast.Name):
+                if node.exc.func.id in STANDARD_NON_TYPE_ERROR_IDS:
+                    self._mark_violation(node.exc.func)
+        elif isinstance(node.exc, ast.Name):
+            if node.exc.id in STANDARD_NON_TYPE_ERROR_IDS:
+                self._mark_violation(node.exc)
+
+    def visit_If(self, node: ast.If) -> None:
+        if self._is_checking_type(node):
+            for stm in ast.iter_child_nodes(node):
+                if isinstance(stm, ast.Raise):
+                    self._check_is_raise_other_than_typeerror(stm)
+        else:
```

Comment:
I think this `else` block is not required at all.
I'll run some tests locally later, and give feedback.

Reply:
Indeed, instead of an `else`, invoke `generic_visit` again:

```py
    def visit_If(self, node: ast.If) -> None:
        if self._is_checking_type(node):
            for stm in ast.iter_child_nodes(node):
                if isinstance(stm, ast.Raise):
                    self._check_is_raise_other_than_typeerror(stm)

        self.generic_visit(node) # <- Here
```

Reply:
Good catch. I ended up moving the recursive call into `_is_checking_type()` instead so that we could make sure all conditional branches involve some kind of type-check, as per [@sbrugman's earlier comment](https://github.com/guilatrova/tryceratops/pull/34#discussion_r740425891), before marking a violation.